SpatialGDK/Source/SpatialGDKFunctionalTests/SpatialGDK/SpatialTestNetReceive/SpatialTestNetReceive.cpp

```diff
@@ -0,0 +1,185 @@
+							  TEXT("bRepNotifyCalledBeforePreNet should be the correct value"));
+
+			RequireEqual_Bool(TestActor->Subobject->bPreNetCalledBeforePostNet, true,
+							  TEXT("bPreNetCalledBeforePostNet should be the correct value"));
+			RequireEqual_Bool(TestActor->Subobject->bRepNotifyCalledBeforePostNet, false,
+							  TEXT("bRepNotifyCalledBeforePostNet should be the correct value"));
+
+			RequireEqual_Bool(TestActor->Subobject->bPreNetCalledBeforeRepNotify, true,
+							  TEXT("bPreNetCalledBeforeRepNotify should be the correct value"));
+			RequireEqual_Bool(TestActor->Subobject->bPostNetCalledBeforeRepNotify, true,
+							  TEXT("bPostNetCalledBeforeRepNotify should be the correct value"));
+
+			RequireEqual_Int(TestActor->Subobject->PreNetNumTimesCalled, 1, TEXT("PreNetNumTimesCalled should be the correct value"));
+			RequireEqual_Int(TestActor->Subobject->PostNetNumTimesCalled, 1, TEXT("PostNetNumTimesCalled should be the correct value"));
+			RequireEqual_Int(TestActor->Subobject->RepNotifyNumTimesCalled, 1, TEXT("RepNotifyNumTimesCalled should be the correct value"));
+
+			FinishStep();
+		},
+		StepTimeLimit);
+}
+
+void ASpatialTestNetReceive::GetLifetimeReplicatedProps(TArray<FLifetimeProperty>& OutLifetimeProps) const
+{
+	Super::GetLifetimeReplicatedProps(OutLifetimeProps);
+
+	DOREPLIFETIME(ASpatialTestNetReceive, TestActor);
+}
+
+ASpatialTestNetReceiveActor::ASpatialTestNetReceiveActor()
+{
+	bReplicates = true;
+
+	Subobject = CreateDefaultSubobject<USpatialTestNetReceiveSubobject>(TEXT("USpatialTestNetReceiveSubobject"));
+	Subobject->SetIsReplicated(true);
+}
+
+void ASpatialTestNetReceiveActor::GetLifetimeReplicatedProps(TArray<FLifetimeProperty>& OutLifetimeProps) const
+{
+	Super::GetLifetimeReplicatedProps(OutLifetimeProps);
+
+	DOREPLIFETIME(ASpatialTestNetReceiveActor, Subobject);
+}
+
+USpatialTestNetReceiveSubobject::USpatialTestNetReceiveSubobject()
+	: Super()
+{
+	SetIsReplicatedByDefault(true);
+
+	TestInt = 0;
+	bOnRepTestInt1Called = false;
+
+	bPreNetReceiveCalled = false;
+	bPostNetReceiveCalled = false;
+
+	bPostNetCalledBeforePreNet = false;
+	bRepNotifyCalledBeforePreNet = false;
+
+	bPreNetCalledBeforePostNet = false;
+	bRepNotifyCalledBeforePostNet = false;
+
+	bPreNetCalledBeforeRepNotify = false;
+	bPostNetCalledBeforeRepNotify = false;
+
+	PreNetNumTimesCalled = 0;
+	PostNetNumTimesCalled = 0;
+	RepNotifyNumTimesCalled = 0;
+}
+
+void USpatialTestNetReceiveSubobject::PreNetReceive()
+{
+	bPreNetReceiveCalled = true;
+
+	bPostNetCalledBeforePreNet = bPostNetReceiveCalled;
+	bRepNotifyCalledBeforePreNet = bOnRepTestInt1Called;
+
+	++PreNetNumTimesCalled;
+}
+
+void USpatialTestNetReceiveSubobject::PostNetReceive()
+{
+	bPostNetReceiveCalled = true;
+
+	bPreNetCalledBeforePostNet = bPreNetReceiveCalled;
+	bRepNotifyCalledBeforePostNet = bOnRepTestInt1Called;
+
+	++PostNetNumTimesCalled;
+}
+
+void USpatialTestNetReceiveSubobject::OnRep_TestInt1(int32 OldTestInt1)
+{
+	bOnRepTestInt1Called = true;
```

Comment:
Why both a count and a boolean?
You could instead:
```
RequireCompare_Int(TestActor->Subobject->RepNotifyNumTimesCalled, EComparisonMethod::Greater_Than_Or_Equal_To, 1, TEXT("RepNotifyNumTimesCalled should be called at least once"));
```

Reply:
this makes more sense, changed!